pkg/agent/git/clone.go

```diff
@@ -57,11 +57,11 @@ func CloneRepository(config *config.Config, project *serverapiclient.Project, au
 }
 
 func shouldCloneBranch(project *serverapiclient.Project) bool {
-	if project.Repository.Branch == nil {
+	if project.Repository.Branch == nil || *project.Repository.Branch == "" {
```

Comment:
This case should not be handled here, a nil-check should be enough.

I can see why you added, when parsing the url below you used `var branch string` which defaults the branch to `""`. This is a mistake since branch will always be assigned in that case. You should use `var branch *string`.